airdrop/job/eligibility.py

```diff
@@ -0,0 +1,95 @@
+        for window in all_windows:
+            details = {}
+            details["airdrop_id"] = window["airdrop_id"]
+            details["rewards_processor"] = window["rewards_processor"]
+            self._active_airdrop_window_map[window["airdrop_window_id"]] = details
+        
+    def __batch_insert(self, values, force=False):
+        start = time.process_time()
+        number_of_rows = len(self.__rows_to_insert)
+        if (force and number_of_rows > 0) or number_of_rows >= 50:
+            self._airdrop_db.bulk_query(self.__insert_snapshot, self.__rows_to_insert)
+            self.__rows_to_insert.clear()       
+            print(f"*****{(time.process_time() - start)} seconds. Inserted {number_of_rows} rows")
+        
+        if(len(values) > 0):
+            self.__rows_to_insert.append(tuple(values))
+
+    def __populate_snapshot(self):
+        query = "select aa.wallet_address, aa.balance, ab.staked from " +\
+                "(select wallet_address, sum(amount) as balance from agix_balances " +\
+                "where balance_type <> 'STAKED' group by wallet_address) as aa, " +\
+                "(select wallet_address, amount as staked from agix_balances where balance_type = 'STAKED') as ab " +\
+                "where aa.wallet_address = ab.wallet_address"
+        snapshot = self._balances_db.execute(query)
+        logger.info(f"Snapshot size is {len(snapshot)} holders")
+
+        for row in snapshot:
+            for window in self._active_airdrop_window_map:
+                total = Decimal(row["balance"]) + Decimal(row["staked"])
+                self.__batch_insert([window, row["wallet_address"], row["balance"], row["staked"], total, self._snapshot_guid])
+        self.__batch_insert([], True)
+
+    def process_eligibility(self):
+        self.__populate_state()
+        if len(self._active_airdrop_window_map) == 0:
+            logger.info(f"No airdrop windows to process for")
+            return
+        
+        logger.info(f"Processing eligibility for windows {self._active_airdrop_window_map.keys()}. Snapshot Index is {self._snapshot_guid}")
+        #self.__populate_snapshot()
+        
+        for window in self._active_airdrop_window_map:
+            processor_name = self._active_airdrop_window_map[window]["rewards_processor"]
+            processor_class = locate("airdrop.job.reward_processors."+processor_name)
+            processor = processor_class(self._airdrop_db, self._active_airdrop_window_map[window]["airdrop_id"],window, self._snapshot_guid)
+            processor.process_rewards()
+
+
+@exception_handler(SLACK_HOOK=SLACK_HOOK, logger=logger)
+def process_eligibility(event, context):      
+    logger.info(f"Processing eligibility")
+
+    e = EligibilityProcessor()
+    e.process_eligibility()
+    
+    logger.info(f"Completed Processing eligibility")
+    return generate_lambda_response(
+        200,
+        "Success"
+    )
+
+
+process_eligibility(None, None)
```

Comment:
Line 95 is for local testing. If we are going to run it as lambda then we don't need this line.

Reply:
Yeah, this is incomplete. Will clean this up when I complete the logic